backend/core/schedule/views.py

```diff
@@ -177,6 +185,43 @@ def generate_link(self, request: AuthedRequest, team_pk: str) -> Response:
 
         return Response(get_cal_settings(request=request, team_pk=team_pk))
 
+    @action(detail=False, methods=["GET"])
+    def next_open(self, request: AuthedRequest, team_pk: str) -> Response:
+        with connection.cursor() as cursor:
+            weekday = request.query_params["day"]
+            now = request.query_params["now"]
+            day_number = {
+                "Sunday": (0,),
+                "Monday": (1,),
+                "Tuesday": (2,),
+                "Wednesday": (3,),
+                "Thursday": (4,),
+                "Friday": (5,),
+                "Saturday": (6,),
+                "Weekday": (1, 2, 3, 4, 5),
+                "Weekend": (0, 6),
+            }[weekday]
+
+            cursor.execute(
+                """
+SELECT
+  gen_date::date "date"
+FROM
+  generate_series(%(now)s::date + '1 day'::interval, (%(now)s::date + '3 years'::interval), '1 day') "gen_date"
+  LEFT JOIN core_scheduledrecipe ON core_scheduledrecipe."on" = gen_date
+    AND team_id = %(team_id)s
+WHERE
+  date_part('dow', gen_date) in %(day_number)s
+  AND id IS NULL
+ORDER BY
+  gen_date ASC
+LIMIT 1;
+""",
+                {"day_number": day_number, "team_id": team_pk, "now": now},
```

Comment:
We aren't properly restricting access to this team here are we?

Shouldn't we check if the user has access to this team before making this query?

Reply:
nvm, the `IsTeamMember` permission covers this. DRF is too magic